### PR TITLE
Added GCM support for Initial Resources

### PR DESCRIPTION
**Pull request by piosz**
part of #13589

### CODE REVIEW

**Comment by piosz** [comment]
@vishh could you please take a look if the communication with GCM looks reasonable?

File: plugin/pkg/admission/initialresources/gcm.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+func retrieveRawSamples(res *gcm.ListTimeseriesResponse, output *[]int) {
+	for _, ts := range res.Timeseries {
+		for _, p := range ts.Points {
+			*output = append(*output, int(p.DoubleValue))
+		}
+	}
+}
+
+func (s *gcmSource) GetUsagePercentile(kind api.ResourceName, perc int64, image string, exactMatch bool, start, end time.Time) (int64, int64, error) {
```
Comment (mwielgus) [inline plugin/pkg/admission/initialresources/gcm.go]: Can we have some test for this?

**Comment by mwielgus** [comment]
Godeps are broken.

**Comment by piosz** [comment]
It's pretty weird. The same error happens for me on HEAD while running godep restore:
```
pszczesniak@piosz:~/go/src/k8s.io/kubernetes$ godep restore
package github.com/prometheus/client_golang/extraction: cannot find package "github.com/prometheus/client_golang/extraction" in any of:
        /usr/lib/google-golang/src/github.com/prometheus/client_golang/extraction (from $GOROOT)
        /usr/local/google/home/pszczesniak/go/src/github.com/prometheus/client_golang/extraction (from $GOPATH)
godep: restore: exit status 1
package github.com/prometheus/client_golang/model: cannot find package "github.com/prometheus/client_golang/model" in any of:
        /usr/lib/google-golang/src/github.com/prometheus/client_golang/model (from $GOROOT)
        /usr/local/google/home/pszczesniak/go/src/github.com/prometheus/client_golang/model (from $GOPATH)
godep: restore: exit status 1
```
On the other hand hack/verify-godeps.sh terminates immediately on HEAD and fails on this branch. Also the errors seems to be unrelated to my change.

**Comment by piosz** [comment]
Godeps is broken due to https://github.com/prometheus/client_golang/pull/173

**Comment by eparis** [comment]
our options
1) fix e2e to not use prometheus/golang_client/extraction
2) fix godep to not die in this situation
3) turn of the verification

uggggh

**Comment by vishh** [comment]
@piosz: Just curious - why are we adding more storage endpoints to the api-server instead of working on stable APIs first as discussed in #13589?

File: plugin/pkg/admission/initialresources/gcm.go (hunk 1 above)
Comment (piosz) [inline plugin/pkg/admission/initialresources/gcm.go]: I'll try to do add some tests.

**Comment by piosz** [comment]
@vishh if you mean Prediction API this will land eventually (for 1.2 I guess) in Heapster as a part of it. If you mean API which allows the access to historical data I wouldn't expect the one to be completed soon since we don't know yet what our need are.

**Comment by piosz** [comment]
Disabled godeps check temporary to allow shippable/travis pass.

File: plugin/pkg/admission/initialresources/gcm.go (hunk 1 above)
Comment (piosz) [inline plugin/pkg/admission/initialresources/gcm.go]: Will do in a follow-up PR. Added issue which tracks progress on that work #14178

**Comment by mwielgus** [comment]
LGTM, assuming that you remove "Remove this" commit.

**Comment by vishh** [comment]
GCM code LGTM.

**Comment by eparis** [comment]
rebase and it should pass verify-godeps.sh without that commit.

**Comment by piosz** [comment]
rebased and removed temp commit

**Comment by thockin** [comment]
NAK - travis doesn't like godeps changes

**Comment by eparis** [comment]
I think godep is the problem.  I'm looking.

**Comment by eparis** [comment]
Hopefully https://github.com/kubernetes/kubernetes/pull/14198 does it right....

**Comment by eparis** [comment]
(but that has nothing to do with an e2e failure)

**Comment by piosz** [comment]
rebased

**Pull request merged**